STF 1.1 documentation updates (round 1) (#136)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/proc_deleting-the-default-smart-gateways.adoc

[id="deleting-the-default-smart-gateways_{context}"]
= Deleting the default Smart Gateways

When you have configured {ProjectShort} for multiple clouds, you can delete the default Smart Gateways if they are no longer in use. The Service Telemetry Operator can remove `SmartGateway` objects that have been created but are no longer listed in the ServiceTelemetry `clouds` list of objects. Removal of SmartGateway objects that are not defined by the `clouds` parameter can be enabled by setting `cloudsRemoveOnMissing: true` in the `ServiceTelemetry` manifest.

> TIP: If you don't want any Smart Gateways deployed, define an empty clouds object via `clouds: {}`

Reviewer: ```suggestion
TIP: If you do not want any Smart Gateways deployed, define an empty clouds object using the `clouds: {}` parameter.
```

> WARNING: Enabling `cloudsRemoveOnMissing` will remove any manually created `SmartGateway` objects in the current namespace without any possibility to restore. The `cloudsRemoveOnMissing` parameter is therefore disabled by default.

Reviewer: ```suggestion
WARNING: The `cloudsRemoveOnMissing` parameter is disabled by default. If you enable the `cloudsRemoveOnMissing` parameter, you remove any manually created `SmartGateway` objects in the current namespace without any possibility to restore. 
```

.Procedure

> To delete the Smart Gateways, complete the following steps:

Reviewer: ```suggestion

```

. Define your `clouds` parameter with the list of cloud objects to be managed by the Service Telemetry Operator. For more information, see xref:clouds_installing-the-core-components-of-stf[].

. Edit the ServiceTelemetry object and add the `cloudsRemoveOnMissing` parameter:
+
----
apiVersion: infra.watch/v1beta1
kind: ServiceTelemetry
metadata:
  ...
spec:
  ...
  cloudsRemoveOnMissing: true
  clouds:
    ...
----

. Save the modifications.

> . Verify the Smart Gateways have been removed. This can take several minutes while the Operators reconcile the changes:

Reviewer: ```suggestion
. Verify that the Smart Gateways have been removed. This can take several minutes while the Operators reconcile the changes:
```
